### ISSUE TITLE
ingress: Investigate if it's possible to remove dummy endpoint

### ISSUE THREAD

**Issue by sayboras** [cilium/cilium#19262]
The watcher here is only for dummy endpoint owned by Ingress object (label cilium.io/ingress: true). It's just a convenient way to upsert (i.e check from cache then perform update or create accordingly), so I don't think we need to watch endpoints or endpointslice in general. 

There is a discussion to find a way to remove this dummy endpoint as well.

_Originally posted by @sayboras in https://github.com/cilium/cilium/pull/18867#discussion_r817467700_

**Comment by sayboras** [comment]
This is not possible without a lot of changes in the current codebase, no harm to have this dummy endpoint in general though. Closing the issue.

**Comment by arukiidou** [comment]
dummy endpoint is
192.192.192.192:9999(IP: "192.192.192.192",Port: 9999)

**Comment by sayboras** [comment]
We should probably revisit this as part of 1.17 dev cycle.

**Comment by joamaki** [comment]
Will see if I can tackle this as part of the new load-balancing control-plane rewrite.

**Comment by joamaki** [comment]
We've now switched to the new load-balancing control-plane so this dummy endpoint is no longer necessary with v1.18. So we can now remove creating it in v1.19 without having issues with rolling upgrade.

**Comment by sathieu** [comment]
@joamaki Can the dummy endpoint be removed now?

(NB: it's causing warning in ArgoCD, because endpoints are ignored)

**Comment by joamaki** [comment]
Yeah in principle it should now be possible to remove it. Will do a PR.
